.github/pull_request_template.md

```diff
@@ -0,0 +1,8 @@
+Fixes/delivers issue #<issue_number>
+
+This PR contains the following changes:
+- <briefly_explain_changes>
+- 
+
+Checks:
+- [ ] Added `#nullable enable` to all the modified files
```

Comment:
Since this message would be visible to external contributors, it would be good to add a short paragraph to "Contributing" article on why we are doing that and reference there.

Reply:
Updated the relevant section in contributing page.
This also delivers the work needed in issue https://github.com/dotnet/templating/issues/2767

Reply:
Some contributing sections are still outdated:
-  Changing build configurations  - I don't think it works this way
-  Notes about putting dotnet new3 to PATH/symlink doesn't work either
-  Branching information is outdated 

I suggest we remove/update them too (in this or different PR) before closing https://github.com/dotnet/templating/issues/2767

Reply:
Updated.